src/PythonMigrationViewExtension/PythonMigrationViewExtension.cs

```diff
@@ -0,0 +1,138 @@
+﻿using Dynamo.Graph.Nodes;
+using Dynamo.Graph.Workspaces;
+using Dynamo.Logging;
+using Dynamo.Models;
+using Dynamo.PythonMigration.Properties;
+using Dynamo.UI.Prompts;
+using Dynamo.ViewModels;
+using Dynamo.Wpf.Extensions;
+using PythonNodeModels;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows.Media;
+
+namespace Dynamo.PythonMigration
+{
+    public class PythonMigrationViewExtension : IViewExtension
+    {
+        private const string EXTENSION_NAME = "Workspace References";
+        private const string EXTENSION_GUID = "1f8146d0-58b1-4b3c-82b7-34a3fab5ac5d";
+        private const string NOTIFICATION_TITLE = "IronPythonNotification";
+
+        private ViewLoadedParams LoadedParams { get; set; }
+        private DynamoViewModel DynamoViewModel { get; set; }
+        private NotificationMessage IronPythonNotification { get; set; }
```

Comment:
do we need the notification as a property ? from what i remember, you make the object, send the notification and not keep track of the object

Reply:
ah, i see, we were using this to not raise it multiple times. with the new requirement to keep track of which workspaces it's been raised in this will likely turn into a dict or something anyway 😛